### PR TITLE
Introduce PhoneField component

### PR DESCRIPTION
**Pull request by wmurray**
This adds a PhoneField component to anchor-ui's form package. The phone field supports typeahead formatting and US phone number validation.

<details><summary>Screenshot</summary>

Screenshot 2023-03-01 at 5 48 00 PM
</details>

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204067175435082

### CODE REVIEW

File: packages/components/form/__specs__/phone_input.spec.tsx
Comment (wmurray) [inline packages/components/form/__specs__/phone_input.spec.tsx:45]: I didn't add these as helpers yet as they seemed pretty focused on the functionality being tested in this component. If we want to abstract them now, though, that works for me.

File: packages/components/form/src/phone_field/phone_field.tsx
Comment (wmurray) [inline packages/components/form/src/phone_field/phone_field.tsx:49]: I am unsure how this bit should be ported over to work correctly. It does not throw any errors in Storybook, so it seems like it may be off. 

I tinkered a bit with the `form_error` component, but wonder if it could be an issue leveraging `FormField` and `Controller` together.

File: packages/components/form/__specs__/phone_input.spec.tsx
Comment (dgalarza) [inline packages/components/form/__specs__/phone_input.spec.tsx:37]: We've been utilizing UserEvent as recommended in our guides in anchor.

https://testing-library.com/docs/user-event/intro/

File: packages/components/form/src/phone_field/phone_field.tsx
Comment (dgalarza) [inline packages/components/form/src/phone_field/phone_field.tsx:40]: When @tute merges his work in https://github.com/BuoySoftware/anchor-ui/pull/139 we'll want to include the `hint` as well here.

File: packages/components/form/__specs__/phone_input.spec.tsx
Comment (wmurray) [reply packages/components/form/__specs__/phone_input.spec.tsx:37]: My mistake

File: packages/components/form/src/phone_field/phone_field.tsx
Comment (wmurray) [reply packages/components/form/src/phone_field/phone_field.tsx:40]: sounds good. I can hold off for that!

File: packages/components/form/__specs__/phone_input.spec.tsx
Comment (wmurray) [reply packages/components/form/__specs__/phone_input.spec.tsx]: Holy cow this is WAY better. 😂

File: packages/components/form/src/phone_field/phone_field.tsx
Comment (wmurray) [inline packages/components/form/src/phone_field/phone_field.tsx:60]: If I understand this correctly, i18next will look for translations in both `anchorForms` and `forms` namespaces, where anchorForms is the `anchor-ui` default and `forms` would be the user-defined namespace. I had a bit of a time getting this to work correctly, and had tinkered quite a bit before landing on this solution. I wasn't sure if we wanted to have `anchorForms` as a fallback if the field isn't defined in a `forms` namespace, and this followed the pattern of the placeholder in the `useFormField` hook above..

Comment (dgalarza) [reply packages/components/form/src/phone_field/phone_field.tsx:60]: @wmurray if you just return true inside this function will `FieldError` kick in? If so that would be the best way to handle this IMO as it has built in error message fallback set up properly. Take a look!

Comment (wmurray) [reply packages/components/form/src/phone_field/phone_field.tsx:60]: I think translations as a whole are botched for this field because I also just noticed that the placeholder isn't working correctly in Storybook.

Comment (wmurray) [reply packages/components/form/src/phone_field/phone_field.tsx:60]: @dgalarza is it worth exploring using the `useFormScope` hook, or is that essentially going to be restating what we have with the initial translation hook? I did something a little like:

```
const { tNamespace } = useFormScope();
const { t } = useTranslation(["anchorForms", "forms"]);
const {t: invalidT} = useTranslation(tNamespace);

....

if (!valid) {
     return invalidT("errors.invalidPhoneNumber") || undefined;
}
```

**Comment by dgalarza** [comment]
@wmurray I tried this locally and it appears to work:

```git
diff --git a/packages/components/form/__specs__/phone_field.spec.tsx b/packages/components/form/__specs__/phone_field.spec.tsx
index 4a677d1..0d51c9d 100644
--- a/packages/components/form/__specs__/phone_field.spec.tsx
+++ b/packages/components/form/__specs__/phone_field.spec.tsx
@@ -9,7 +9,7 @@ import { act, render, waitFor, UserRenderResult } from "../test_utils";
 describe("<PhoneField />", () => {
   const translations = {
     errors: {
-      invalidPhoneNumber: "This phone number is invalid",
+      phoneNumber: "This phone number is invalid",
     },
     hints: {
       test_form: {
@@ -68,7 +68,7 @@ describe("<PhoneField />", () => {
   context("with an error", () => {
     it("displays errors associated to the field", async () => {
       const { user, getByLabelText, getByText } = setup();
-      const errorMessage = i18n.t("forms:errors.invalidPhoneNumber");
+      const errorMessage = i18n.t("forms:errors.phoneNumber");
 
       await act(() => {
         user.click(getByText("Submit"));
diff --git a/packages/components/form/src/phone_field/phone_field.tsx b/packages/components/form/src/phone_field/phone_field.tsx
index a469343..e861ea4 100644
--- a/packages/components/form/src/phone_field/phone_field.tsx
+++ b/packages/components/form/src/phone_field/phone_field.tsx
@@ -52,15 +52,8 @@ export const PhoneField: React.FC<PhoneFieldProps> = ({
   const label = labelProp ?? formFieldLabel;
   const inputId = snakeCase(name);
 
-  const phoneNumberValidation = (phoneNumber: string): string | undefined => {
-    const valid = isPossiblePhoneNumber(phoneNumber, "US");
-
-    if (!valid) {
-      return (
-        t("errors.invalidPhoneNumber", { ns: ["anchorForms", "forms"] }) ||
-        undefined
-      );
-    }
+  const phoneNumberValidation = (phoneNumber: string): boolean => {
+    return isPossiblePhoneNumber(phoneNumber, "US");
   };
 
   return (
```

**Comment by wmurray** [comment]
@dgalarza worked for me as well, now. Not sure why it was giving me a hard time about the boolean last week. 🤷

**Review by dgalarza — APPROVED** [review]

**Review by acostanzo — APPROVED** [review]
Looks great to me! I really dig the user event stuff.

**Pull request merged**
